src/snowflake/snowpark/session.py

```diff
@@ -1834,17 +1834,15 @@ def update_query_tag(self, tag: dict) -> None:
                     f"Expected query tag to be valid json. Current query tag: {tag_str}"
                 )
 
-    def table(
-        self, name: Union[str, Iterable[str]], suppress_ast: bool = False
-    ) -> Table:
+    def table(self, name: Union[str, Iterable[str]], emit_ast: bool = True) -> Table:
```

Comment:
nit: table should be public API? so `_emit_ast` according to PR description?

Reply:
Done.